### PR TITLE
Adds collections management API

### PR DESCRIPTION
**Pull request by Abdulazeez-98**
Hi,

I have been using your library for a personal project. Thank you for the time and effort you put into it. I really like how it's written.

I noticed that it doesn't include the collections API. I tried my best to implement them in this library while trying to code with the same used style. I based this branch on feat/v2, because it seemed to be the most updated one.
I wrote a basic test suite. I honestly do not write many tests so I hope it's not that bad. I tested the code manually and it worked as well. In the pull request, the test suite is "skipped" (.skip) because they need an API key and will therefore fail in the build phase.

Have a great day!

### CODE REVIEW

**Comment by dan-online** [comment]
Okay so @Abdulazeez-98 you'll need to rebase with `feat/v2` because I added a *lot* of tests and tried out a lot of the API, the good news though is that you'll be able to take out the .skip as I've added my keys to the CI

**Comment by Abdulazeez-98** [comment]
Okay @dan-online I rebased it on feat/v2. Shall I make a new pull request or this one is properly updated?

File: tests/Bunny.test.ts [hunk 1]
```diff
@@ -223,3 +223,74 @@ describe('BunnyCdnStream', () => {
...
   });
 });
+
+describe(
+  'Collections API 2E2',
+  () => {
+    const stream = new BunnyCdnStream({ videoLibrary: process.env.BUNNY_VIDEO_LIBRARY || '123', apiKey: process.env.BUNNY_API_KEY || '123' });
+    let createdCollectionGUID: string;
+
+    test('Fetching a non-existing collection throws an error', async () => {
```
Comment (dan-online) [inline tests/Bunny.test.ts:233]: Could you format these like the other tests, for example "GIVEN library w/o collection THEN throw error"

File: tests/Bunny.test.ts [hunk 2]
```diff
@@ -223,3 +223,74 @@ describe('BunnyCdnStream', () => {
...
+      expect(response.success).toBe(true);
+      expect(response.message).not.toBeUndefined();
+      expect(response.statusCode).not.toBeUndefined();
+    });
+
+    test('Collection was indeed deleted', async () => {
+      await expect(stream.getCollection(createdCollectionGUID)).rejects.toThrowError();
+    });
+  },
+  { timeout: 10000 }
```
Comment (dan-online) [inline tests/Bunny.test.ts:295]: Is this required for collections?

**Review by dan-online — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -343,6 +343,88 @@ export class BunnyCdnStream {
...
+   * await stream.getCollection("0273f24a-79d1-d0fe-97ca-b0e36bed31es")
+   * ```
+   */
+  public async getCollection(collectionId: string): Promise<BunnyCdnStream.GetCollectionResponse> {
+    const options = this.getOptions();
+    options.url += `/library/${this.options.videoLibrary}/collections/${collectionId}`;
+    return this.request<BunnyCdnStream.GetCollectionResponse>(options, 'getCollection');
+  }
+
+  /**
```
[the remark this answers is not in the document]
Comment (Abdulazeez-98) [reply src/index.ts:425]: Okay I will do it.

File: tests/Bunny.test.ts [hunk 4]
```diff
@@ -223,3 +223,74 @@ describe('BunnyCdnStream', () => {
...
+    const stream = new BunnyCdnStream({ videoLibrary: process.env.BUNNY_VIDEO_LIBRARY || '123', apiKey: process.env.BUNNY_API_KEY || '123' });
+    let createdCollectionGUID: string;
+
+    test('Fetching a non-existing collection throws an error', async () => {
+      await expect(stream.getCollection('non-existing-GUID')).rejects.toThrowError();
+    });
+
+    test('Creating a collection returns a valid response', async () => {
+      const response = await stream.createCollection('TestCollection');
+      expect(response.videoLibraryId).not.toBeUndefined();
```
[the remark this answers is not in the document]
Comment (Abdulazeez-98) [reply tests/Bunny.test.ts]: Yes, it is way better than my code.

File: tests/Bunny.test.ts (hunk 1 above)
Comment (Abdulazeez-98) [reply tests/Bunny.test.ts]: Okay

File: tests/Bunny.test.ts (hunk 2 above)
Comment (Abdulazeez-98) [reply tests/Bunny.test.ts]: I just copied it from your last tests (I think you then removed it).
I will remove it.

**Review by dan-online — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**